src/Catalyst.Core.Modules.Consensus/Deltas/Building/DeltaBuilder.cs

```diff
@@ -0,0 +1,103 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.Collections.Generic;
+using Autofac.Features.AttributeFilters;
+using Catalyst.Abstractions.Consensus;
+using Catalyst.Abstractions.Consensus.Deltas;
+using Catalyst.Abstractions.Cryptography;
+using Catalyst.Abstractions.Hashing;
+using Catalyst.Abstractions.Kvm;
+using Catalyst.Abstractions.P2P;
+using Catalyst.Protocol.Peer;
+using Catalyst.Protocol.Wire;
+using LibP2P;
+using Nethermind.Store;
+using Serilog;
+
+namespace Catalyst.Core.Modules.Consensus.Deltas.Building
+{
+    /// <inheritdoc />
+    public sealed class DeltaBuilder : IDeltaBuilder
+    {
+        private readonly IDeltaTransactionRetriever _transactionRetriever;
+        private readonly IDeterministicRandomFactory _randomFactory;
+        private readonly IHashProvider _hashProvider;
+        private readonly PeerId _producerUniqueId;
+        private readonly IDeltaCache _deltaCache;
+        private readonly IDateTimeProvider _dateTimeProvider;
+        private readonly IStateProvider _stateProvider;
+        private readonly IDeltaExecutor _deltaExecutor;
+        private readonly ILogger _logger;
+
+        public DeltaBuilder(IDeltaTransactionRetriever transactionRetriever,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Constructor has 9 parameters, which is greater than the 7 authorized.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)